### DISCUSSION TITLE
FluxCD upgrade to v2.0.0.rc.1

### DISCUSSION THREAD

**Discussion by KomanRudden** [fluxcd/flux2#3870] [General] [answered]
Good morning,

We have setup a GitHub Action for Flux and we recently upgraded from v0.x to v2.0.0.rc.1 successfully through a PR.

A couple of weeks later a colleague asked me why there were Kustomizations on older apiVersions. Here is an example:
![image](https://github.com/fluxcd/flux2/assets/1030623/6b2aa2c0-aab1-4ef0-8c83-a93ad5678f47)

Here is an example of an upgraded Kustomization:
![image](https://github.com/fluxcd/flux2/assets/1030623/de76f717-3b0d-48cc-9a15-aff592c2c610)

I then went about updating the apiVersions manually and it was then that Flux died completely. flux-system pods terminated, kustomizations clearly wouldn't reconcile, apps managed by Flux were deleted e.g. pods, Ambassador Mappings.

After trying unsuccessfully to revert apiVersion changes we eventually did a flux uninstall and bootstrap. 

We are on Azure AKS. 

The only other comment is that we did the upgrade of Flux a few weeks back but we migrated the Flux repo from Bitbucket to GitHub three months ago. Everything after the Bitbucket->Github migration worked just fine. ImageAutoUpdates, reconciles, everything worked 100%. Only when we edited the apiVersions did it crash.

What is confusing me on the [release documentation](https://github.com/fluxcd/flux2/releases/tag/v2.0.0-rc.1) is this where it specifies this
![image](https://github.com/fluxcd/flux2/assets/1030623/f85266e6-7908-4a05-8563-ae49a0878d70)

This is what I did but it clealy caused alot of issues. Do you have any suggestions or comments please. My system is back and running so this is just a discussion to understand it all better.

Thank you

**Comment by stefanprodan** [comment, marked as answer]
You're confusing Kustomize configs with Flux APIs. `kustomize.config.k8s.io` is not a Flux thing, if you changed the version there, then it's like deleting the whole thing.

**Reply by hiddeco** [reply]
See also https://github.com/fluxcd/flux2/issues/321 for more historic background.

**Reply by KomanRudden** [reply]
Oh wow, as soon as you said that the light went on. It's k8s.io and not fluxcd.io. I was focused on the kind: Kustomization only. 

Thank you for the link. I'm glad to see this has been confusing other mere mortals as well.

Thank to you both.
